packages/node_modules/@ciscospark/spark-core/src/interceptors/request-event.js

```diff
@@ -0,0 +1,93 @@
+/**!
+ *
+ * Copyright (c) 2015-2017 Cisco Systems, Inc. See LICENSE file.
+ * @private
+ */
+
+import {get} from 'lodash';
+import {Interceptor} from '@ciscospark/http-core';
+
+/**
+ * @class
+ */
+export default class RequestEventInterceptor extends Interceptor {
+  /**
+   * @returns {RequestEventInterceptor}
+   */
+  static create() {
+    return new RequestEventInterceptor({spark: this});
+  }
+
+  /**
+   * @see Interceptor#onRequest
+   * @param {Object} options
+   * @returns {Object}
+   */
+  onRequest(options) {
+    const logger = get(this, `spark.logger`, console);
+    try {
+      this.spark.trigger(`request:start`, options);
+    }
+    catch (error) {
+      logger.info(`RequestEventInterceptor.onRequest failed for options:`, options);
```

Comment:
The error string should be something more like `logger.error('event handler for "request:start" failed', error)`